lightning/src/ln/channelmanager.rs

```diff
@@ -11230,19 +11230,30 @@ This indicates a bug inside LDK. Please report this error at https://github.com/
 							insert_short_channel_id!(short_to_chan_info, chan);
 						}
 
-						let mut pending_events = self.pending_events.lock().unwrap();
-						pending_events.push_back((
-							events::Event::ChannelReady {
-								channel_id: chan.context.channel_id(),
-								user_channel_id: chan.context.get_user_id(),
-								counterparty_node_id: chan.context.get_counterparty_node_id(),
-								funding_txo: Some(
-									splice_promotion.funding_txo.into_bitcoin_outpoint(),
-								),
-								channel_type: chan.funding.get_channel_type().clone(),
-							},
-							None,
-						));
+						{
+							let mut pending_events = self.pending_events.lock().unwrap();
+							pending_events.push_back((
+								events::Event::ChannelReady {
+									channel_id: chan.context.channel_id(),
+									user_channel_id: chan.context.get_user_id(),
+									counterparty_node_id: chan.context.get_counterparty_node_id(),
+									funding_txo: Some(
+										splice_promotion.funding_txo.into_bitcoin_outpoint(),
+									),
+									channel_type: chan.funding.get_channel_type().clone(),
+								},
+								None,
+							));
+							splice_promotion.discarded_funding.into_iter().for_each(
+								|funding_info| {
+									let event = Event::DiscardFunding {
+										channel_id: chan.context.channel_id(),
+										funding_info,
+									};
```

Comment:
Could you update the `DiscardFunding` docs to indicate when it is emitted, similarly to those of `ChannelReady`?